### ISSUE TITLE
truecolor messes up background color after redraw

### ISSUE THREAD

**Issue by git-bruh** [termbox/termbox2#51]
```c
#define TB_IMPL
#define TB_LIB_OPTS
#define TB_OPT_TRUECOLOR
#include "termbox.h"

int main(void) {
    struct tb_event ev;

    tb_init();
    tb_set_output_mode(TB_OUTPUT_TRUECOLOR);

    for (size_t i = 0; i < 3; i++) {
        tb_clear();
        tb_printf(0, 0, 0xFF0000, 0x00FF00, "hello from termbox");
        tb_present();

        tb_poll_event(&ev);
    }

    tb_poll_event(&ev);
    tb_shutdown();

    return 0;
}
```

Initial draw:
![IMG_20221128_012220](https://user-images.githubusercontent.com/67585967/204156696-c81f71a0-e872-440f-8e7f-bb184e720bb5.jpg)

After resize:
![IMG_20221128_012234](https://user-images.githubusercontent.com/67585967/204156708-fe7fed99-f8ff-4e9e-aa64-87b570832271.jpg)

**Comment by adsr** [comment]
Good catch! `TB_DEFAULT` and `TB_TRUECOLOR_DEFAULT` were recently changed to `0x2000` and `0x2000000` to allow for black (`0`) in 256-color and true-color modes. Global clear attrs (`global.(fg|bg)`) default to `TB_DEFAULT` and persist even after a `tb_set_output_mode` change, so `0x2000` gets interpretted as dark green instead of "default".

Some possible remediations:

1. Document that callers should ensure sane `tb_set_clear_attrs` after calling `tb_set_output_mode`.

2. Add a new function `tb_set_output_mode_and_clear_attrs`.

3. Modify `tb_set_output_mode` to include clear attr params. (ABI break)
 
4. Do best effort translation of `global.(fg|bg)` after a mode change.

5. Go back to `0` as default in all modes, which I think is more intuitive. For black, we could introduce special high-bit macros `TB_256_BLACK` and `TB_TRUECOLOR_BLACK`.

I'm leaning toward 1 and 5 for now, maybe 2. What do you think @git-bruh?

If we didn't care about space we could make separate `tb_cell` fields for all attrs including `is_default` and/or always use 32-bit ints for `(fg|bg)`. This would reduce some but not all complexity surrounding mode changes. I'm personally not in favor of this as it would harm a few library users that need `tb_cell` to remain small.

**Comment by adsr** [comment]
Tagging @mitchellwrosen who may have an opinion on this.

In #41 we changed `TB_DEFAULT` and `TB_TRUECOLOR_DEFAULT` to `0x2000` and `0x2000000` to allow for black (0) in those modes. Solution 5 mentioned above would go back to `TB_DEFAULT==0` in all modes, which is conceptually simpler and would prevent at least one instance of the bug git-bruh found here. It also eliminates the weirdness of `<any color> | TB_DEFAULT` being equivalent to `TB_DEFAULT`. The downside is that library users would need to use `TB_256_BLACK` and `TB_TRUECOLOR_BLACK` for black. Any thoughts?

**Comment by git-bruh** [comment]
Why not implicitly call `tb_set_clear_attrs` from `tb_set_output_mode` itself? Like

```diff
diff --git a/termbox.h b/termbox.h
index febf364..d3a793b 100644
--- a/termbox.h
+++ b/termbox.h
@@ -1660,17 +1660,22 @@ int tb_set_output_mode(int mode) {
     switch (mode) {
         case TB_OUTPUT_CURRENT:
             return global.output_mode;
+#ifdef TB_OPT_TRUECOLOR
+        case TB_OUTPUT_TRUECOLOR:
+            tb_set_clear_attrs(TB_TRUECOLOR_DEFAULT, TB_TRUECOLOR_DEFAULT);
+            break;
+#endif
         case TB_OUTPUT_NORMAL:
         case TB_OUTPUT_256:
         case TB_OUTPUT_216:
         case TB_OUTPUT_GRAYSCALE:
-#ifdef TB_OPT_TRUECOLOR
-        case TB_OUTPUT_TRUECOLOR:
-#endif
-            global.output_mode = mode;
-            return TB_OK;
+            tb_set_clear_attrs(TB_DEFAULT, TB_DEFAULT);
+            break;
+        default:
+            return TB_ERR;
     }
-    return TB_ERR;
+    global.output_mode = mode;
+    return TB_OK;
 }
 
 int tb_peek_event(struct tb_event *event, int timeout_ms) {
```

**Comment by adsr** [comment]
The original bug is fixed in 1eaa24b. I went with the solution described in https://github.com/termbox/termbox2/issues/51#issuecomment-1328366870.

> Why not implicitly call tb_set_clear_attrs from tb_set_output_mode itself? ...

After thinking about it, it's not just the clear attrs but all attrs that are potentially broken after a mode switch. For example, if you start in `TB_OUTPUT_NORMAL` with some cells colored `TB_CYAN`, then switch to `TB_OUTPUT_GRAYSCALE`, the previous cyan cells will eventually be re-interpreted as a shade of gray when the front buffer is cleared (after a resize). (The new function `tb_invalidate` offers a way for callers to make that happen immediately instead of waiting for a resize.) The only way I can think of to solve this properly is with a `tb_cell.mode` field, but IMO that's too complex for what is probably an edge case.
